### PR TITLE
Ddcnls 3833:Add regex validation

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/controllers/nonUKReg/BusinessRegController.scala [hunk 1]
```diff
@@ -58,6 +58,7 @@ trait BusinessRegController extends BackLinkController {
   def send(service: String, businessType: String) = AuthAction(service).async { implicit bcContext =>
     BusinessRegistrationForms.validateCountryNonUKAndPostcode(businessRegistrationForm.bindFromRequest, service, bcContext.user.isAgent).fold(
       formWithErrors => {
+        println(formWithErrors)
```
Comment (Gaur4vGaur) [inline app/controllers/nonUKReg/BusinessRegController.scala]: Get rid of println

File: app/utils/BusinessCustomerConstants.scala [hunk 2]
```diff
@@ -31,3 +31,24 @@ object BusinessCustomerConstants {
...
+  val length105 = 105
+
+  val postcodeRegex = "^[A-Z]{1,2}[0-9][0-9A-Z]?\\s?[0-9][A-Z]{2}|BFPO\\s?[0-9]{1,10}$"
+  val lineRegex = "^[A-Za-z0-9 \\-,.&']{1,35}$"
+  val utrRegex = """^[0-9]{10}$"""
+  val nameRegex = "^[a-zA-Z &`\\-\'^]{1,35}$"
+  val businessNameRegex = "^[a-zA-Z0-9 '&\\\\/]{1,105}$"
+  val idNumberRegex = "^[a-zA-Z0-9 '&\\-]{1,60}$"
+  val issuingInstitutionRegex = "^[a-zA-Z0-9 '&\\-\\/]{1,40}$"
+  val issuingCountryRegex = "(?!^GB$)^[A-Z]{2}$"
```
Comment (Gaur4vGaur) [inline app/utils/BusinessCustomerConstants.scala]: Please follow naming conventions to declare constants

File: test/forms/BusinessRegistrationFormsSpec.scala [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+ */
+
+package forms
+
+import org.scalatestplus.play.{OneAppPerSuite, PlaySpec}
+import models.{Address, BusinessRegistration}
+
+class BusinessRegistrationFormsSpec extends PlaySpec with OneAppPerSuite {
+
+  "Businessegistration form" should {
```
Comment (Gaur4vGaur) [inline test/forms/BusinessRegistrationFormsSpec.scala]: Use 'must' for consistency

Comment (Gaur4vGaur) [reply test/forms/BusinessRegistrationFormsSpec.scala]: please use must throughout

File: test/forms/BusinessRegistrationFormsSpec.scala [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+          formWithErrors.errors(0).message must be("Address line 2 cannot be more than 35 characters")
+          formWithErrors.errors.length must be(1)
+        },
+        success => {
+          fail("Form should give an error")
+        }
+      )
+    }
+
+    // Error message needs to be updated
```
Comment (Gaur4vGaur) [inline test/forms/BusinessRegistrationFormsSpec.scala]: is this comment still valid?

File: test/forms/BusinessRegistrationFormsSpec.scala [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+  "overseasCompanyForm form" should {
+
+    "throw overseasCompanyForm error on not electing hasBusinessUniqueId" in {
+      BusinessRegistrationForms.overseasCompanyForm.bind(Map[String,String]()).fold(
+        formWithErrors => {
+          formWithErrors.errors.head.message must be("You must select one option")
+          formWithErrors.errors.length must be(1)
+        },
+        _ => {
+
```
Comment (Gaur4vGaur) [inline test/forms/BusinessRegistrationFormsSpec.scala:371]: would it be worth calling fail, to catch the situation when form does not throw any validation?

File: test/forms/BusinessVerificationFormsSpec.scala [hunk 6]
```diff
@@ -0,0 +1,553 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import org.scalatestplus.play.{OneAppPerSuite, PlaySpec}
+
+class BusinessVerificationFormsSpec extends PlaySpec with OneAppPerSuite {
+  "Soletrader form " should {
```
Comment (Gaur4vGaur) [inline test/forms/BusinessVerificationFormsSpec.scala]: Please be consistent and use 'must' throughout

**Review by Gaur4vGaur — CHANGES_REQUESTED** [review]

File: test/forms/BusinessRegistrationFormsSpec.scala (hunk 4 above)
Comment (ashwinirudrappa) [reply test/forms/BusinessRegistrationFormsSpec.scala]: Fixed

File: test/forms/BusinessVerificationFormsSpec.scala (hunk 6 above)
Comment (ashwinirudrappa) [reply test/forms/BusinessVerificationFormsSpec.scala]: Fixed

File: app/utils/BusinessCustomerConstants.scala (hunk 2 above)
Comment (ashwinirudrappa) [reply app/utils/BusinessCustomerConstants.scala]: Fixed

File: test/forms/BusinessRegistrationFormsSpec.scala (hunk 5 above)
Comment (ashwinirudrappa) [reply test/forms/BusinessRegistrationFormsSpec.scala:371]: Fixed

File: test/forms/BusinessRegistrationFormsSpec.scala (hunk 3 above)
Comment (ashwinirudrappa) [reply test/forms/BusinessRegistrationFormsSpec.scala]: Fixed

File: test/forms/BusinessVerificationFormsSpec.scala [hunk 7]
```diff
@@ -0,0 +1,553 @@
...
+          formWithErrors.errors.length must be(1)
+        },
+        success => {
+          fail("Form should give an error")
+        }
+      )
+    }
+  }
+
+  "limitedPartnershipForm" should {
```
Comment (Gaur4vGaur) [inline test/forms/BusinessVerificationFormsSpec.scala]: Use 'must' for consistency

File: test/forms/BusinessRegistrationFormsSpec.scala [hunk 8]
```diff
@@ -0,0 +1,439 @@
...
+ */
+
+package forms
+
+import org.scalatestplus.play.{OneAppPerSuite, PlaySpec}
+import models.{Address, BusinessRegistration}
+
+class BusinessRegistrationFormsSpec extends PlaySpec with OneAppPerSuite {
+
+  "Businessegistration form" must {
```
Comment (Gaur4vGaur) [inline test/forms/BusinessRegistrationFormsSpec.scala]: typo!

**Review by Gaur4vGaur — CHANGES_REQUESTED** [review]

File: test/forms/BusinessRegistrationFormsSpec.scala (hunk 8 above)
Comment (ashwinirudrappa) [reply test/forms/BusinessRegistrationFormsSpec.scala]: Fixed it, so sorry

**Review by Gaur4vGaur — APPROVED** [review]

**Pull request closed without merging**